From c68e889e59fff4f6cc56cc4cb3bc63629dff4cf7 Mon Sep 17 00:00:00 2001
From: Sean Nyekjaer <sean@nyekjaer.dk>
Date: Sat, 29 Jul 2017 12:44:11 +0200
Subject: [PATCH 12/12] arm64: dts: marvell: armada-3720-espressobin: Add vqmmc
 regulator for SD slot

By adding this regulator, the SD cards are usable at higher speed
protocols such as SDR104.

This patch was tested with an SDHC card compatible with UHS-I.

Signed-off-by: Sean Nyekjaer <sean@nyekjaer.dk>
---
 arch/arm64/boot/dts/marvell/armada-3720-espressobin.dts | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-espressobin.dts b/arch/arm64/boot/dts/marvell/armada-3720-espressobin.dts
index 409385c4195a..f9756193e78a 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-espressobin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-espressobin.dts
@@ -60,6 +60,19 @@
 		device_type = "memory";
 		reg = <0x00000000 0x00000000 0x00000000 0x20000000>;
 	};
+
+	vcc_sd_reg1: regulator {
+		compatible = "regulator-gpio";
+		regulator-name= "vcc_sd1";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <3300000>;
+
+		gpios = <&gpionb 4 GPIO_ACTIVE_HIGH>;
+		gpios-states = <0>;
+		states = <1800000 0x1
+			  3300000 0x0>;
+		enable-active-high;
+	};
 };
 
 /* J9 */
@@ -82,7 +95,7 @@
 	wp-inverted;
 	bus-width = <4>;
 	cd-gpios = <&gpionb 3 GPIO_ACTIVE_LOW>;
-	no-1-8-v;
+	vqmmc-supply = <&vcc_sd_reg1>;
 	marvell,pad-type = "sd";
 	status = "okay";
 };
-- 
2.13.3

